src/librustdoc/html/render.rs

```diff
@@ -648,43 +644,94 @@ fn concise_compared_strs(s1: &str, s2: &str) -> (String, String) {
+    fn extract_attr(value: &str) -> &str {
+        let mut iter = value.rsplitn(2, '-');
+        if let (Some(n), Some(tag)) = (iter.next(), iter.next()) {
+            if n.parse::<usize>().is_ok() {
+                tag
+            } else {
+                value
+            }
+        } else {
+            value
+        }
+    }
+
+    if let (Some(id1), Some(id2)) = (attrs1.get("id"), attrs2.get("id")) {
+        let tag1 = extract_attr(id1);
+        let tag2 = extract_attr(id2);
+
+        tag1 == tag2
+    } else {
+        !attrs1.contains_key("id") && !attrs2.contains_key("id")
+    }
+}
+
+
+fn print_message(msg: &str, intro_msg: &mut bool, span: &Span, text: &str) {
+    if !*intro_msg {
+        println!("WARNING: documentation for this crate may be rendered \
+                  differently using the new Pulldown renderer.");
+        println!("    See https://github.com/rust-lang/rust/issues/44229 for details.");
+        *intro_msg = true;
+    }
+    println!("WARNING: rendering difference in `{}`", concise_str(text));
+    println!("   --> {}:{}:{}", span.filename, span.loline, span.locol);
+    println!("{}", msg);
+}
+
+fn render_difference(diff: &html_diff::Difference, intro_msg: &mut bool, span: &Span, text: &str) {
     match *diff {
         html_diff::Difference::NodeType { ref elem, ref opposite_elem } => {
-            println!("    {} Types differ: expected: `{}`, found: `{}`",
-                     elem.path, elem.element_name, opposite_elem.element_name);
+            print_message(&format!("    {} Types differ: expected: `{}`, found: `{}`",
+                                   elem.path, elem.element_name, opposite_elem.element_name),
+                          intro_msg, span, text);
         }
         html_diff::Difference::NodeName { ref elem, ref opposite_elem } => {
-            println!("    {} Tags differ: expected: `{}`, found: `{}`",
-                     elem.path, elem.element_name, opposite_elem.element_name);
+            print_message(&format!("    {} Tags differ: expected: `{}`, found: `{}`",
+                                   elem.path, elem.element_name, opposite_elem.element_name),
+                          intro_msg, span, text);
         }
         html_diff::Difference::NodeAttributes { ref elem,
-                                     ref elem_attributes,
-                                     ref opposite_elem_attributes,
-                                     .. } => {
-            println!("    {} Attributes differ in `{}`: expected: `{:?}`, found: `{:?}`",
-                     elem.path, elem.element_name, elem_attributes, opposite_elem_attributes);
+                                                ref elem_attributes,
+                                                ref opposite_elem_attributes,
+                                                .. } => {
+            if !check_attributes(elem_attributes, opposite_elem_attributes) {
```

Comment:
...i thought we agreed that `check_attributes` didn't need to be there any more `>_>`

Reply:
Ah my bad, completely misunderstood. ><